### PR TITLE
Create sitemap

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
#Create sitemap

create a sitemap in order to get better indexing
let's see how it's the best format for sitemaps for google (i think xml w the last update times) and create an endpoint in the backend to generate it.

fixes https://github.com/EveripediaNetwork/issues/issues/771

### CODE REVIEW

File: src/Sitemap/data/staticPagesData.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const staticPagesData = [
```
Comment (kesar) [inline src/Sitemap/data/staticPagesData.ts:1]: sitemaps are intented to be indexed by google. Private pages or 404 pages are not needed.
and of course all the wikis are the most important part of the sitemap 👍🏻 
also worth adding categories (tags can be skipped for now)

Comment (Emmanueldmlr) [reply src/Sitemap/data/staticPagesData.ts:1]: Okay then. I decided to start with the static pages though...I need to make some database queries that I'm not familiar with right now. So, I plan to reach out to @AshakaE or try to read up about it.

**Comment by Emmanueldmlr** [comment]
image

File: src/Sitemap/controllers/sitemap.controller.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const [wikisIds, categoriesId] = await Promise.all([
+      this.wikiService.wikisIds(),
+      this.categoryService.categoriesIds(),
+    ])
+    wikisIds.map(wiki =>
+      smStream.write({
+        url: `/wiki/${wiki.id}`,
+        changefreq: 'daily',
+        priority: 1,
+        lastmod: this.lastmod,
```
Comment (kesar) [inline src/Sitemap/controllers/sitemap.controller.ts:50]: lastmod you can get it from db

File: src/Sitemap/controllers/sitemap.controller.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      }),
+    )
+    categoriesId.map(category =>
+      smStream.write({
+        url: `/categories/${category.id}`,
+        changefreq: 'weekly',
+        priority: 1,
```
Comment (kesar) [inline src/Sitemap/controllers/sitemap.controller.ts:57]: lower priority

File: src/Sitemap/data/staticPagesData.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  '/activity',
+  '/blog',
+  '/categories',
+  '/create-wiki',
```
Comment (kesar) [inline src/Sitemap/data/staticPagesData.ts:5]: private

File: src/Sitemap/data/staticPagesData.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const staticPagesData = [
+  '/activity',
```
Comment (kesar) [inline src/Sitemap/data/staticPagesData.ts:2]: probably we want to assign different changefreq & priority to this pages.

most of them its fine w a changefreq weekly and priority of 0.7

activity should be hourly and 1
blog daily and 0.8
/ hourly and 1

File: src/Sitemap/controllers/sitemap.controller.ts (hunk 2 above)
Comment (Emmanueldmlr) [reply src/Sitemap/controllers/sitemap.controller.ts]: done

File: src/Sitemap/controllers/sitemap.controller.ts (hunk 3 above)
Comment (Emmanueldmlr) [reply src/Sitemap/controllers/sitemap.controller.ts]: done

File: src/Sitemap/data/staticPagesData.ts (hunk 5 above)
Comment (Emmanueldmlr) [reply src/Sitemap/data/staticPagesData.ts]: done

File: src/Sitemap/data/staticPagesData.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const staticPagesData = [
+  { url: '/activity', priority: 1, changeFreq: 'hourly' },
+  { url: '/blog', priority: 0.8, changeFreq: 'daily' },
+  { url: '/categories', priority: 0.7, changeFreq: 'weekly' },
+  { url: '/create-wiki', priority: 0.7, changeFreq: 'weekly' },
```
Comment (kesar) [inline src/Sitemap/data/staticPagesData.ts:5]:
```suggestion
```

**Pull request merged**